# CMSE 491: Bioinformatics and Computational Biology

* [Description](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#description)
* [Instructor Contact Information](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#instructor-contact-information)
* [Course Outline and Materials](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#course-outline-and-materials)
* [Schedule, Location, Calendar, and Office Hours](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#schedule-location-calendar-and-office-hours)
* [Website and Communication](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#website-and-communication)
* [Course Activities](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#course-activities)
* [Grading Information](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#grading-information)
* [Attendance, Conduct, Honesty, and Accommodations](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#attendance-conduct-honesty-accomodations)


## Description
This course is an introduction to contemporary topics in bioinformatics and computational biology, dealing with combining large-scale data and modern analytical techniques to gain biological/biomedical insights. In each topic, centered around a recent paper, we will discuss the major biological & biomedical questions, explore the relevant molecular/genomics/biomedical datasets, and understand the underlying statistical, probabilistic, & machine-learning approaches.

Students will learn how to formulate problems for quantitative inquiry, design computational projects, understand and think critically about data & methods, communicate research findings, perform reproducible research, and practice open science. Students will apply all these by carrying out a group project, presenting their project in class, and submit a report at the end of the course.

**Note**  
Open to both undergraduate and graduate students. Counts toward the CMSE minor, graduate certificates, and dual PhD. Please email Heather Johnson at john1451@msu.edu for an override.

**Prerequisites**  
CMSE 201 and two semesters of introductory biology (LB 144 and 145 OR BS 161 and 162 OR BS 181H and 182H, or equivalent).
Statistics at the level of STT 231 is strongly recommended. Basically, it would be assumed that you:
- know how to code in one of the mainstream languages like Python and R,
- have an understanding of basic statistics and probability, and
- have studied basic genetics, molecular biology, and cellular biology.

## Instructor Contact Information
Arjun Krishnan | ...
:------------ | :------------
Affiliation | Dept. Computational Mathematics, Science, and Engineering</br>Dept. Biochemistry and Molecular Biology
Office | 2507H Engineering Building
Contact | Email: arjun@msu.edu</br>Twitter: [@compbiologist](https://twitter.com/compbiologist)</br>Website: https://cmse.msu.edu/directory/faculty/arjun-krishnan/

\[ [Top](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#cmse-491-bioinformatics-and-computational-biology) ]

## Course Outline and Materials
### Major Topics
_(subject to changes)_
* Genome assembly and annotation
* Sequence alignment and pattern finding
* Comparative genomics
* Genetic variation and quantitative genetics
* Regulatory genomics
* Functional genomics
* Molecular and digital evolution
* Molecular docking and dynamics
* Protein structure prediction
* Modeling cellular pathways
* Metabolomics
* Large-scale biological networks

Each topic is covered over two classes, a "Lecture" and a "Paper discussion".

### Recommended Materials
TBD

\[ [Top](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#cmse-491-bioinformatics-and-computational-biology) ]

## Schedule, Location, Calendar, and Office Hours
S/L | Info
:------------ | :------------
Schedule | Mon and Wed</br>8:30-9:50 am
Location | 202 Urban Planning & Land. Arch. Building

### Calendar
This calendar contains the class schedule and the links to the lecture slides and reading materials. Download the PDF file [here](https://github.com/krishnanlab/teaching/blob/master/compbio/Schedule.pdf).

Week | Date | Topic | Content | Learning Materials
:----- | ----------: | :-- | :------------- | :--
Week 1 | Mon, Jan 08 | Introduction and Overview | Course overview | In-coming survey \[[Link](https://goo.gl/forms/Vt0tFiakqVB42Mmo2)]</br></br>**Lecture1** \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Lectures/Lecture1.pdf)]</br></br>**Required reading:**</br>- All biology is computational biology \[[Journal](http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.2002050)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Markowetz2017_all-bio-is-compbio.pdf)]</br>- Computational biologists: moving to the driver's seat \[[Journal](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1357-1)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Yanai2017_compbiologist-drivers-seat.pdf)]
Week 1 | Wed, Jan 10 | Introduction and Overview | Getting started | **Lecture2** \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Lectures/Lecture2.pdf)]</br></br>**Required reading:**</br>- So, you want to be a computational biologist \[[Journal](https://www.nature.com/articles/nbt.2740)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Loman2013_so-you-want-to-be-a-combiologist.pdf)]
Week 2 | Mon, Jan 15 | No class (MLK Day) | |
Week 2 | Wed, Jan 17 | Genome assembly and annotation | Lecture | **Lecture3** \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Lectures/Lecture3.pdf)]
Week 3 | Mon, Jan 22 | Genome assembly and annotation | Paper discussion | Velvet: Algorithms for de novo short read assembly using de Bruijn graphs \[[Journal](http://genome.cshlp.org/content/18/5/821)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Zerbino2008_velvet-assembler.pdf)]
Week 3 | Wed, Jan 24 | Sequence alignment and pattern finding | Lecture | **Lecture4** \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Lectures/Lecture4.pdf)]</br></br>**Optional reading:**</br>- [BLAST Wikipedia page](https://en.wikipedia.org/wiki/BLAST)</br>- [BLAST Glossary](https://www.ncbi.nlm.nih.gov/books/NBK62051/)</br>- [BLAST Practical details](https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=BlastHelp)</br>- [Statistics of sequence similarity scores](https://www.ncbi.nlm.nih.gov/BLAST/tutorial/Altschul-1.html)
Week 4 | Mon, Jan 29 | Sequence alignment and pattern finding | Paper discussion | Basic local alignment search tool \[[Journal](https://doi.org/10.1016/S0022-2836(05)80360-2)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Altschul1990_BLAST.pdf)]</br></br>**Additional reading:**</br>-Rapid and sensitive protein similarity searches \[[Journal](http://science.sciencemag.org/content/227/4693/1435)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Lipman-Pearson1985_fastp.pdf)]</br>- [FASTA (the algorithm) Wikipedia page](https://en.wikipedia.org/wiki/FASTA)</br>- An introduction to sequence similarity ("homology") searching \[[Journal](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3820096/)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Pearson2013_intro-seq-similarity-searching.pdf)]</br>- [Substitution matrix Wikipedia page](https://en.wikipedia.org/wiki/Substitution_matrix)</br>- Selecting the right similarity-scoring matrix \[[Journal](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3848038/)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Pearson2013_selecting-the-right-scoring-matrix.pdf)]
Week 4 | Wed, Jan 31 | Comparative genomics | Lecture | TBD</br></br>**Additional reading:**</br>- Comparative genomics as a tool to understand evolution and disease \[[Journal](http://genome.cshlp.org/content/23/7/1063.full)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Alfo%CC%88ldi-Lindblad-Toh2013_comparative-genomics-evolution-disease.pdf)]</br>- Dissecting evolution and disease using comparative vertebrate genomics \[[Journal](https://www.nature.com/articles/nrg.2017.51)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Meadows-Lindblad-Toh2017_comparative-genomics-vertebrate-evolution-disease.pdf)]
Week 5 | Mon, Feb 05 | Comparative genomics | Paper discussion | Whole-genome alignment:</br>- MUMmer1: Alignment of whole genomes \[[Journal](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC148804/)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Delcher1999_MUMmer1.pdf)]</br>- MUMmer2: Fast algorithms for large-scale genome alignment and comparison \[[Journal](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC117189/)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Delcher2002_MUMmer2.pdf)]</br>- MUMmer3: Versatile and open software for comparing large genomes \[[Journal](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2004-5-2-r12)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Kurtz2004_MUMmer3.pdf)]</br>- MUMmer4: A fast and versatile genome alignment system \[[Journal](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005944)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Marc%CC%A7ais2018_MUMmer4.pdf)]
Week 5 | Wed, Feb 07 | Genetic variation and quantitative genetics | Lecture | TBD
Week 6 | Mon, Feb 12 | Genetic variation and quantitative genetics | Paper discussion | Genome-wide association studies and multiple hypothesis correction:</br>- Mining Genome-Wide Genetic Markers \[[Journal](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002828)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Zhang2012_mining-genomewide-genetic-markers.PDF)]</br>- Genome-Wide Association Studies \[[Journal](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002822)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Bush-Moore2012_GWAS.PDF)]</br>- How does multiple testing correction work? \[[Journal](https://www.nature.com/articles/nbt1209-1135)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Noble2009_primer-multiple-testing-correction.pdf)]</br>- Statistical significance for genomewide studies \[[Journal](http://www.pnas.org/content/100/16/9440.full)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Storey-Tibshirani2003_statistical-significance-genomewide-studies.pdf)]
Week 6 | Wed, Feb 14 | Regulatory genomics | Lecture | **Lecture7** \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Lectures/Lecture7.pdf)]
Week 7 | Mon, Feb 19 | Regulatory genomics | Paper discussion | - What are DNA sequnence motifs? \[[Journal](https://www.nature.com/articles/nbt0406-423)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Dhaeseleer2006_what-are-dna-motifs.pdf)]</br>- How does DNA sequence motif discovery work? \[[Journal](https://www.nature.com/articles/nbt0806-959)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Dhaeseleer2006_how-does-motif-discovery-work.pdf)]</br>- What is the Expectation Maximization algorithm? \[[Journal](https://www.nature.com/articles/nbt1406)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Do-Batzoglou2008_what-is-EM.pdf)]</br>- Practical Strategies for Discovering Regulatory DNA Sequence Motifs \[[Journal](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.0020036)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/MacIsaac-Fraenkel2006_strategies-for-discovering-motifs.pdf)]
Week 7 | Wed, Feb 21 | Functional genomics | Lecture | **Lecture8** \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Lectures/Lecture8.pdf)]
Week 7 | Mon, Feb 26 | Functional genomics | Paper discussion | TBD
Week 8 | Wed, Feb 28 | Mini Primers | TBD | TBD
Week 8 | Mon, Mar 05 | No class (Spring break) | | 
Week 9 | Wed, Mar 07 | No class (Spring break) | | 
Week 9 | Mon, Mar 12 | Mid-term project proposal presentations | |
Week 10 | Wed, Mar 14 | Molecular and digital evolution | Lecture | **Lecture9** \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Lectures/Lecture9.pdf)]
Week 10 | Mon, Mar 19 | Molecular and digital evolution | Paper discussion | TBD
Week 11 | Wed, Mar 21 | Molecular docking and dynamics | Lecture | **Lecture10** \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Lectures/Lecture10.pdf)]
Week 11 | Mon, Mar 26 | Molecular docking and dynamics | Paper discussion | - Predicting protein structures with a multiplayer online game \[[Journal](https://www.nature.com/articles/nature09304)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Cooper2010_Foldit.pdf)]</br>- [GROMACS tutorial](http://md.chem.rug.nl/~mdcourse/index.html)
Week 12 | Wed, Mar 28 | Protein structure prediction | Lecture | **Lecture11** \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Lectures/Lecture11.pdf)]
Week 12 | Mon, Apr 02 | Protein structure prediction | Paper discussion | - Evolutionarily conserved networks of residues mediate allosteric communication in proteins \[[Journal](https://www.nature.com/articles/nsb881)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Suel2003_networks-of-residues.pdf)]</br></br>**Additional reading:**</br>- Inferring Pairwise Interactions from Biological Data Using Maximum-Entropy Probability Models \[[Journal](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004182)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Stein2015_maximum-entropy-probability-models.pdf)]</br>- Gleaning structural and functional information from correlations in protein multiple sequence alignments \[[Journal](https://www.sciencedirect.com/science/article/pii/S0959440X16300355)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Neuwald2016_structure-function-from-msa.pdf)]
Week 13 | Wed, Apr 04 | Modeling cellular pathways | Lecture | **Lecture12** \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Lectures/Lecture12.pdf)]
Week 13 | Mon, Apr 09 | Modeling cellular pathways | Paper discussion | Construction of a genetic toggle switch in Escherichia coli \[[Journal](https://www.nature.com/articles/35002131)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Gardner2000-Ecoli-genetic-toggle-switch.pdf)]</br></br>**Additional reading:**</br>- Networks dynamics and cell physiology \[[Journal](https://www.nature.com/articles/35103078)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Tyson2001_network-dynamics-cell-physiology.pdf)]</br>- Computational studies of gene regulatory networks: in numero molecular biology \[[Journal](https://www.nature.com/articles/35066056)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Hasty2001_innumero-molecular-biology.pdf)]</br>- Network motifs: theory and experimental approaches \[[Journal](https://www.nature.com/articles/nrg2102)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Alon2007_network-motifs.pdf)]
Week 14 | Wed, Apr 11 | Metabolomics | Lecture | **Lecture13** \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Lectures/Lecture13.pdf)]
Week 14 | Mon, Apr 16 | Metabolomics | Paper discussion | - Network-based prediction of human tissue-specific metabolism \[[Journal](https://www.nature.com/articles/nbt.1487)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Shlomi2008_tissue-specific-metabolic-network.pdf)]</br>- Integration of expression data in genome-scale metabolic network reconstructions (Mini supplementary reading to help with the main paper) \[[Journal](https://www.frontiersin.org/articles/10.3389/fphys.2012.00299/full)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Blazier2012_integrating-expression-with-metabolic-network.pdf)]</br></br>**Additional reading:**</br>- A protocol for generating a high-quality genome-scale metabolic reconstruction \[[Journal](https://www.nature.com/articles/nprot.2009.203)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Thiele2010_protocol-for-metabolic-reconstruction.pdf)]</br>- What is flux balance analysis? \[[Journal](https://www.nature.com/articles/nbt.1614)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Orth2010_what-is-fba.pdf)]</br>- Applications of genome‐scale metabolic reconstructions \[[Journal](http://msb.embopress.org/content/5/1/320.short)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Oberhardt2009_applications-of-metabloic-reconstructions.pdf)]
Week 15 | Wed, Apr 18 | Large-scale biological networks | Lecture | **Lecture14** \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Lectures/Lecture14.pdf)]
Week 15 | Mon, Apr 23 | Large-scale biological networks | Paper discussion | Genomic analysis of regulatory network dynamics reveals large topological changes \[[Journal](https://www.nature.com/articles/nature02782)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Luscombe2004_topological-network-dynamics.pdf)]</br></br>**Additional reading:**</br>- Network biology: understanding the cell's functional organization \[[Journal](https://www.nature.com/articles/nrg1272)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Barabasi2004_network-biology.pdf)]</br>- Network medicine: a network-based approach to human disease \[[Journal](https://www.nature.com/articles/nrg2918)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Barabasi2011_network-medicine.pdf)]</br>- Learning biological networks: from modules to dynamics \[[Journal](https://www.nature.com/articles/nchembio.122)] \[[PDF](https://github.com/krishnanlab/teaching/blob/master/compbio/Papers/Bonneau2008_learning-networks.pdf)]</br>- Inferring cellular networks – a review \[[Journal](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-8-S6-S5)] \[[PDF](https://bmcbioinformatics.biomedcentral.com/track/pdf/10.1186/1471-2105-8-S6-S5)]
Week 16 | Wed, Apr 25 | Final project presentations 1 | |
Week 16 | Mon, Apr 30 | Final project presentations 2 | |

#### Project deadlines
Item | Due date
:--- | -------:
Project profile | Mon, Jan 29
Project topic/team | Wed, Feb 07
Project pre-proposal | Wed, Feb 14
Project proposal | Mon, Feb 26
Proposal reviews | Mon, Mar 05
Mid-term project proposal presentations | Mon, Mar 12
Review response | Mon, Mar 14
Mid-course project report | Wed, Apr 04
Final project report | Wed, Apr 25
Final project presentations 1 | Wed, Apr 25
Final project presentations 2 | Mon, Apr 30

### Office Hours
Wednesday 5-7pm

I will block this time from my schedule and be present in my office.

Couple of things to note:
1. While I'm happy to chat with you in person, many times, just sending me a message on Slack with your questions/concerns might work as well. So, if you have specific Qs in mind, just shoot me a message and let's see if we can resolve it then and there.
2. If you would indeed like to meet in person, please try to meet me during this time. But, don't worry if you can't make it during this window for some reason. Again, just send me a message on Slack and we'll find a time that works for both of us.

\[ [Top](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#cmse-491-bioinformatics-and-computational-biology) ]

## Website and Communication

### Course website
This [GitHub repo](https://github.com/krishnanlab/teaching/tree/master/compbio) will serve as the course website.  

### Communication
The primary mode of communication in this course (including major announcements), will be the course Slack account https://cmse491bioinfocompbio.slack.com. All of you should have invitations to join this account in your MSU email.  

**Emails**  
Although the bulk of the communication will take place via Slack, at times (rarely), we will send out important course information via email. This email is sent to your MSU email address (the one that ends in “@msu.edu”). You are responsible for all information sent out to your University email account, and for checking this account on a regular basis.

\[ [Top](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#cmse-491-bioinformatics-and-computational-biology) ]

## Course Activities

### Pre-class Assignments
For each topic, you will be assigned a paper after the topic's "lecture" class that you are required to read. The link to the PDF of the paper will be posted on this page next to the topic on the [Calendar](https://github.com/krishnanlab/teaching/tree/master/compbio#calendar) along with instructions on a specific analysis in the paper you should pay special attention to.

You are required to turn-in a two-page report containing the following:

**Part 1** (page 1) of your report: Analysis and summary of the paper as a whole.
It should contain the following three sections:
1. _Background_
    - What problem/task is the paper trying to address? What is the biological problem and/or the computational challenge?
    - How was the problem/task addressed up to that point and what were the limitations of the _then_ current practice?
2. _Summary of contributions_
    - What is the contribution of this paper in terms of approach, algorithmic techniques, computational ideas?
    - Summarize the conclusion(s) your draw from each main figure in the paper. Based on your conclusions, are the major claims of the paper justified?
3. _Limitations and Future directions_
    - What are the limitations of this paper or open questions that arise from this paper?
    - What are your suggestions for addressing these limitations?

**Part 2** (page 2) of your report: Critical and thorough analysis of a specific section/figure.
It should contain the following four sections:
1. _Data_
    - What data did they use to perform the analysis presented in the figure?
    - Where did each piece of data come from?
2. _Methods_
    - What techniques and algorithms did they use?
    - Does the paper have a detailed description of how to perform this analysis, enough for someone to repeat that analysis?
    - Does the paper have source code to reproduce the presented results?
3. _Evaluation_
    - What are the measures/metrics plotted for each figure panel?
    - How is the success of each analysis evaluated?
4. _Conclusions_
    - Are the conclusions you draw from the figure in agreement with those drawn by the authors?

This report is due _before_ the topic's "Paper discussion" class. Points will be deducted for reports shorter or longer than two-pages. Wrong papers will be graded as zero.

### Class Participation
In general:
- Do the pre-class assignments and additional readings.
- Show up to class.
- Work in groups during in-class discussion sessions.
- No one will have the perfect background: **Ask questions** about computational or biological concepts.
- Correct me when I am wrong.

#### Paper discussion
You will also take turns to present the assigned paper during each topic's "Paper discussion" class. Make sure you sign-up.
- Two students together present each paper.
- The presentation should focus on the computational/analytical parts, not necessarily on detailed biological background & conclusions.
    1. What is the problem the authors are trying to solve? [description of the problem along with why it is important]
    2. What are their claims about the _then_ current practices and their limitations? [existing approaches to solve the problem & their pros-and-cons]
    3. What’s their approach? What’s new in it and what is their rationale for it being potentially successful? [description of the *new ideas*, their *merits* in comparison to existing ones, and *rationale*]
    4. What are the major contributions and limitations of this paper?
    5. What are some open questions and next steps (for addressing the limitations)?
- The two students will also make a note of all the points discussed in the class during the presentation, write-up them up by working with me, and post the discussion on [PubPeer](https://pubpeer.com/).

### Scribing
Each topic's "Lecture" class will have two dedicated scribes who will take notes on the lecture, work with Arjun to refine the notes, and circulate a final version to the rest of the class.
- The scribes should submit their _individually completed_ drafts of their scribe notes within 3 days after lecture. I will read those notes and give comments/suggestions.
- The two scribes should then work together to combine their drafts+comments into a single final scribe notes and submit within 6 days after lecture.

### Semester Project and Presentation
A major goal of this course is to prepare you for performing original research in computational biology, and for effectively presenting your ideas and research. The semester project will serve as the most practical way to do exactly that.

Students are encouraged to form teams of two and work on this project by leveraging their complementary skills. Projects can take any one of the following flavors:
- Design and implement a new computational method for a task in biology
- Improve an existing method
- Perform an evaluation of several existing methods
- Develop a fully-reproducible documentation and codebase for an existing analysis in a paper

The outcomes of this semester-long project should include:
1. Well-documented code to:
    - download and process the data
    - perform the computational analysis and generate all the results
    - visualize the results as various plots
2. Detailed final report containing the following sections:
    - Abstract
    - Introduction
    - Data and Methods
    - Results and Discussion
    - Limitations and Future Directions
    - References
3. A presentation (slides) that describes your project - motivation, exact problem, approach, results, discussion & conclusions, limitations & future direcrtions, acknowledgements.

There are several [project deadlines](https://github.com/krishnanlab/teaching/tree/master/compbio#project-deadlines) throughout the course that will help you stay on track, allowing you to complete a substantial project.
1. Describe your previous research, areas of research interest in bioinformatics / computational-biology, type of project that best fits your interests. Post this description in a profile that lets your classmates know you and find potential partners. **Project profile due Mon, Jan 29.**
2. Discuss with Arjun (and any other PI), read recent papers, talk to potential partners. Describe project ideas and form groups. **Project topic/team due Wed, Feb 07.**
3. Prepare a two-page pre-proposal (Page1: text; Page2: figures & references). **Project pre-proposal due Wed, Feb 14.**
4. Write 4-page proposal describing project goals, division of work, milestones, datasets, and challenges. **Project proposal due Mon, Feb 26.**
5. Review and discuss proposals (NIH review format). **Reviews due Mon, Mar 05.**
6. Address peer evaluations, revise aims, scope, list of final goals & deliverables. **Response due Mon, Mar 14** (note: due after presentation, not Mar 12).
7. Continue making substantial progress on proposed milestones. Write outline/first-draft of final report. Meet Arjun to discuss all results and get feedback on the draft. **Mid-course project report due Wed, Apr 04.**
8. Complete milestones, finalize results, figures, write-up in conference publication format. As part of the report, comment on your overall project experience. **Final project report due Wed, Apr 25.**
9. Final project presentations will take place on the **Wed, Apr 25** and **Mon, Apr 30**.

After you have formed teams, I will create private channels for each team on Slack so that you can effectively communicate with each other.

\[ [Top](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#cmse-491-bioinformatics-and-computational-biology) ]

## Grading Information
Activity | Percentage
:----- | ---------:
Pre-class assignments | ~35%
Class participation | ~15%
Scribing | ~10%
Project | ~40%

### Grading Scale
Point | Percentage
----: | ---------:
4.0 | ≥ 90%
3.5 | ≥ 85%
3.0 | ≥ 80%
2.5 | ≥ 75%
2.0 | ≥ 70%
1.5 | ≥ 65%
1.0 | ≥ 60%
0.0 | < 60%

**Note:** Grades will not be curved. Your grade is based on your own effort and progress, not based on competition with your classmates.

\[ [Top](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#cmse-491-bioinformatics-and-computational-biology) ]

## Attendance, Conduct, Honesty, and Accommodations

### Class Attendance
This class is heavily based on material presented and worked on in class, and it is critical that you attend and participate fully every week! Therefore, class attendance is absolutely required. An unexcused absence will result in zero points for the day. Arriving late or leaving early without prior arrangement with the instructor of your session counts as an unexcused absence. Note that if you have a legitimate reason to miss class (such as job, graduate school, or medical school interviews), you must arrange this ahead of time to be excused from class. Three unexcused absences will result in the reduction of your grade by one step (e.g., from 4.0 to 3.5), with additional absences reducing your grade further at the discretion of the course instructor.

### Code of Conduct
All conduct should serve the singular goal of sustain a friendly, supportive, and fun environment where we can do our best work and have a great time doing it.
* Do work that you’re proud of, from the smallest piece of code to the entire project.
* Be supportive of your classmates; respect each others' strengths, weaknesses, differences, and beliefs.
* Communicate openly & respectfully with everyone in the class.
* Ask for help; at the same time, respect and appreciate others' time and effort.

Respectful and responsible behavior is expected at all times, which includes not interrupting other students, turning your cell phone off, refraining from non-course-related use of electronic devices, and not using offensive or demeaning language in our discussions. Flagrant or repeated violations of this expectation may result in ejection from the classroom, grade-related penalties, and/or involvement of the university Ombudsperson.

I am unequivocally dedicated to providing a harassment-free experience for everyone, regardless of gender, gender identity and expression, age, sexual orientation, disability, physical appearance, body size, race, or religion (or lack thereof). We will not tolerate harassment of colleagues in any form. Behaviors that could be considered discriminatory or harassing, or unwanted sexual attention, will not be tolerated and will be immediately reported to the appropriate MSU office (which may include the MSU Police Department).

### Academic honesty
Intellectual integrity is the foundation of the scientific enterprise. In all instances, you must do your own work and give proper credit to all sources that you use in your papers and oral presentations – any instance of submitting another person's work, ideas, or wording as your own counts as plagiarism. This includes failing to cite any direct quotations in your essays, research paper, class debate, or written presentation. The MSU College of Natural Science adheres to the policies of academic honesty as specified in the General Student Regulations 1.0, Protection of Scholarship and Grades, and in the all-University statement on Integrity of Scholarship and Grades, which are included in Spartan Life: Student Handbook and Resource Guide. Students who plagiarize will receive a 0.0 in the course. In addition, University policy requires that any cheating offense, regardless of the magnitude of the infraction or punishment decided upon by the professor, be reported immediately to the dean of the student's college.

It is important to note that **plagiarism in the context of this course includes, but is not limited to**, directly copying another student's solutions to in-class or homework problems; copying materials from online sources, textbooks, or other reference materials without citing those references in your source code or documentation, or having somebody else do your pre-class work, in-class work, or homework on your behalf. Any work that is done in collaboration with other students should state this explicitly, and have their names as well as yours listed clearly.

More broadly, we ask that students adhere to the Spartan Code of Honor academic pledge, as written by the Associated Students of Michigan State University (ASMSU): "As a Spartan, I will strive to uphold values of the highest ethical standard. I will practice honesty in my work, foster honesty in my peers, and take pride in knowing that honor is worth more than grades. I will carry these values beyond my time as a student at Michigan State University, continuing the endeavor to build personal integrity in all that I do."

### Accomodations
If you have a university-documented learning difficulty or require other accommodations, please provide me with your VISA as soon as possible and speak with me about how I can assist you in your learning.  If you do not have a VISA but have been documented with a learning difficulty or other problems for which you may still require accommodation, please contact MSU’s Resource Center for People with Disabilities (355-9642) in order to acquire current documentation.

\[ [Top](https://github.com/krishnanlab/teaching/blob/master/compbio/README.md#cmse-491-bioinformatics-and-computational-biology) ]
